pkg/pool-utils/contracts/BaseMinimalSwapInfoPool.sol

```diff
@@ -44,8 +44,9 @@ abstract contract BaseMinimalSwapInfoPool is IMinimalSwapInfoPool, BasePool {
 
         if (request.kind == IVault.SwapKind.GIVEN_IN) {
             // Fees are subtracted before scaling, to reduce the complexity of the rounding direction analysis.
-            request.amount = _subtractSwapFeeAmount(request.amount);
+            request.amount = _subtractSwapFeeAmount(request.amount, request.userData);
 
+            emit SwapFeePercentageChanged(getSwapFeePercentage(request.userData, OperationType.SWAP));
```

Comment:
This is being emitted in both sides of the if-else clause. Maybe move it to right after the call to `_beforeSwapJoinExit()` at the top?